connexion/decorators/parameter.py

```diff
@@ -87,10 +89,13 @@ def parameter_to_arg(parameters, function):
     def wrapper(*args, **kwargs):
         logger.debug('Function Arguments: %s', arguments)
 
-        try:
-            request_body = flask.request.json
-        except exceptions.BadRequest:
-            request_body = None
+        if 'application/json' in consumes:
```

Comment:
This is not the right check. There are lots of "+json" mimetypes that have to be parsed as json